test/e2e/storage/testsuites/volume_modify.go

```diff
@@ -0,0 +1,310 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	storagev1alpha1 "k8s.io/api/storage/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/discovery"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/feature"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	e2evolume "k8s.io/kubernetes/test/e2e/framework/volume"
+	storageframework "k8s.io/kubernetes/test/e2e/storage/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+const (
+	modifyPollInterval               = 2 * time.Second
+	setVACWaitPeriod                 = 30 * time.Second
+	modifyingConditionSyncWaitPeriod = 2 * time.Minute
+	modifyVolumeWaitPeriod           = 10 * time.Minute
+	vacCleanupWaitPeriod             = 30 * time.Second
+)
+
+type volumeModifyTestSuite struct {
+	tsInfo storageframework.TestSuiteInfo
+}
+
+// InitCustomVolumeModifyTestSuite returns volumeModifyTestSuite that implements TestSuite interface
+// using custom test patterns
+func InitCustomVolumeModifyTestSuite(patterns []storageframework.TestPattern) storageframework.TestSuite {
+	return &volumeModifyTestSuite{
+		tsInfo: storageframework.TestSuiteInfo{
+			Name:         "volume-modify",
+			TestPatterns: patterns,
+			SupportedSizeRange: e2evolume.SizeRange{
+				Min: "1Gi",
+			},
+			TestTags: []interface{}{feature.VolumeAttributesClass},
+		},
+	}
+}
+
+// InitVolumeModifyTestSuite returns volumeModifyTestSuite that implements TestSuite interface
+// using testsuite default patterns
+func InitVolumeModifyTestSuite() storageframework.TestSuite {
+	patterns := []storageframework.TestPattern{
+		storageframework.DefaultFsDynamicPV,
+		storageframework.BlockVolModeDynamicPV,
+		storageframework.NtfsDynamicPV,
+	}
+	return InitCustomVolumeModifyTestSuite(patterns)
+}
+
+func (v *volumeModifyTestSuite) GetTestSuiteInfo() storageframework.TestSuiteInfo {
+	return v.tsInfo
+}
+
+func (v *volumeModifyTestSuite) SkipUnsupportedTests(driver storageframework.TestDriver, pattern storageframework.TestPattern) {
+	_, ok := driver.(storageframework.VolumeAttributesClassTestDriver)
+	if !ok {
+		e2eskipper.Skipf("Driver %q does not support VolumeAttributesClass tests - skipping", driver.GetDriverInfo().Name)
+	}
+	if !driver.GetDriverInfo().Capabilities[storageframework.CapBlock] && pattern.VolMode == v1.PersistentVolumeBlock {
```

Comment:
Why skip it? What's the relationship between the `block` volume mode and volume modification feature?

Reply:
I included block volumes in the VAC e2e patterns (`storageframework.BlockVolModeDynamicPV`), because I figured that drivers may modify filesystem vs block volumes differently (especially CSI drivers) and thus it would be valuable to test both.

However, some drivers don't support block volumes so the block tests must be skipped in those cases. This was copied from the volume expansion test cases which does the same thing here: https://github.com/kubernetes/kubernetes/blob/9e2075b3c87061d25759b0ad112266f03601afd8/test/e2e/storage/testsuites/volume_expand.go#L98-L100

Reply:
Please add a comment at L86. It will help others to understand the reason. 

Maybe we need to decouple it from the block volume mode in the future? The vac feature doesn't have this constraint in my mind. Please add a to-do item.

Reply:
Added comment with TODO